test/common/http/http2/codec_impl_test.cc

```diff
@@ -396,7 +396,7 @@ TEST_P(Http2CodecImplTest, Invalid204WithContentLength) {
   // maximum frame size is 16K. We will add 3,000 headers that will take us above this size and
   // not easily compress with HPACK. (I confirmed this generates 26,468 bytes of header data
   // which should contain a continuation.)
-  for (uint i = 1; i < 3000; i++) {
+  for (unsigned i = 1; i < 3000; i++) {
```

Comment:
Can you just make these explicitly sized types? This is the default style in Envoy.

Reply:
That's fine IFF these are envoy args passed to envoy functions alone.

It's a horrible style if consuming stdcxx/absl classes, your intwidth assumptions are either wrong today (Win32 is 32, Unix is 32 or 64) or they become wrong tomorrow (128LP architectures?)

Two things are going on. MSVC is especially picky about confusing unsigned for signed types. And reflectors and emplacement types must match identically. So if the int is going to a stdcxx/absl class, using fixed width types is very ill-advised.

This one could be either 'unsigned int' or 'uint32_t', perhaps. It's only uint that isn't a C[++] type at all.

Reply:
Just to close this discussion, in all of these cases, what is 32 bits on windows and 64 bits on linux can't be used as an argument to a cb function declaration or emplace_back invocation. These types are each correct, tested and re-tested on both linux and windows for correctness according to the class declarations.

Reply:
OK, but please consider that Envoy style remains fixed types, so where we're not forced to do this for interop with STL etc. let's keep things as they are. I haven't spotted any violations of this in this PR, but worth keeping in mind. Resolved.